mirror-main.go

```diff
@@ -170,7 +170,18 @@ func doMirror(sURLs mirrorURLs, progressReader *progressBar, accountingReader *a
 		sURLs.Error = nil
 		return sURLs
 	}
-
+	// If source size <= 5GB we can use server side copy/CopyObject API for s3.
+	if length <= 5*1024*1024*1024 {
+		sourcePath := filepath.ToSlash(filepath.Join(sourceAlias, sourceURL.Path))
```

Comment:
Also filepath.Join(sourceAlias, sourceURL.Path) would only work for FS,  for object storage 'sourceAlias' is not part of the sourcePath. ?